File: src/main/java/graph/edge/ChannelFactory.java
package graph.edge;

import graph.vertex.Vertex;
import org.jetbrains.annotations.NotNull;

public interface ChannelFactory {
    default Edge createChannel(@NotNull Vertex from, @NotNull Vertex to, int weight) {
        return createChannel(from, to, weight, Edge.ERRORS_AMOUNT);
    }

    Edge createChannel(@NotNull Vertex from, @NotNull Vertex to, int weight, double errors);
}

File: src/main/java/mvc/model/channel/ChannelFactory.kt
package mvc.model.channel

import mvc.model.node.Node

interface ChannelFactory {
    fun createChannel(from: Node, to: Node, weight: Int): Channel {
        return createChannel(from, to, weight, Channel.ERRORS_AMOUNT)
    }

    fun createChannel(from: Node, to: Node, weight: Int, errors: Double): Channel
}
